components/brave_wallet_ui/common/slices/endpoints/swap.endpoints.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// types
+import { BraveWallet } from '../../../constants/types'
+import { WalletApiEndpointBuilderParams } from '../api-base.slice'
+
+// utils
+import { handleEndpointError } from '../../../utils/api-utils'
+
+export const swapEndpoints = ({
+  mutation,
+  query
+}: WalletApiEndpointBuilderParams) => {
+  return {
+    generateBraveSwapFee: mutation<
+      {
+        response: BraveWallet.BraveSwapFeeResponse | null
+        errorString: string
+      },
+      BraveWallet.BraveSwapFeeParams
+    >({
+      queryFn: async (params, { endpoint }, extraOptions, baseQuery) => {
+        const { swapService } = baseQuery(undefined).data
+        try {
+          const result = await swapService.getBraveFee(params)
+          return {
+            data: result
+          }
+        } catch (error) {
+          return handleEndpointError(
+            endpoint,
+            'Unable to fetch Brave Swap fee',
+            error
+          )
+        }
+      }
+    }),
+
+    generateSwapQuote: mutation<
```

Comment:
Any good reason for not sticking with the existing convention of using the `get*` prefix?

Reply:
I used the generate* prefix instead of get* for these endpoints to help show that these are mutations instead of queries. I used mutations so that the results can be recomputed when called again instead of returning a cached value. Please let me know if we can use queries/cached values instead